freqtrade/freqtradebot.py

```diff
@@ -393,9 +398,21 @@ def create_trade(self) -> bool:
 
         # running get_signal on historical data fetched
         # to find buy signals
+        if self.config['edge']['enabled']:
+            whitelist = self.edge.filter(whitelist)
+
         for _pair in whitelist:
             (buy, sell) = self.strategy.get_signal(_pair, interval, self.exchange.klines.get(_pair))
             if buy and not sell:
+                stake_amount = self._get_trade_stake_amount(_pair)
+                if not stake_amount:
+                    return False
```

Comment:
what's the reason to check stake-amount per pair? if my stake is 1 ETH - but i only own 0.5 ... that's going to fail for every pair... ?

Reply:
The whole point of Edge Positioning is that you adjust your stake_amount per pair considering its expectancy. if you follow _get_trade_stake_amount you will see that it calls edge.stake_amount which in turn returns the position size for that specific pair. but maybe I use it in a wrong place? the idea being to force stake_amount (position size) per pair ... as well as corresponding stoploss